src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1200,144 +1202,231 @@ private function addRobotsIndexSection(ArrayNodeDefinition $rootNode)
+                                            $config['scope'] = preg_quote(implode('', self::resolveUrl(self::parseUrl('.'), self::parseUrl($config['base_uri']))));
+                                        }
 
-        $subNode = $subNode
+                                        return $config;
+                                    })
                                 ->end()
-                            ->end()
-                        ->end()
-                    ->end()
-                ->end()
-            ->end()
-        ;
-    }
-
-    private function addHttpClientOptionsSection(NodeBuilder $rootNode)
-    {
-        $rootNode
-            ->integerNode('max_host_connections')
-                ->info('The maximum number of connections to a single host.')
-            ->end()
-            ->arrayNode('default_options')
-                ->fixXmlConfig('header')
-                ->children()
-                    ->scalarNode('auth_basic')
-                        ->info('An HTTP Basic authentication "username:password".')
-                    ->end()
-                    ->scalarNode('auth_bearer')
-                        ->info('A token enabling HTTP Bearer authorization.')
-                    ->end()
-                    ->arrayNode('query')
-                        ->info('Associative array of query string values merged with URL parameters.')
-                        ->useAttributeAsKey('key')
-                        ->beforeNormalization()
-                            ->always(function ($config) {
-                                if (!\is_array($config)) {
-                                    return [];
-                                }
-                                if (!isset($config['key'])) {
-                                    return $config;
-                                }
+                                ->validate()
+                                    ->ifTrue(function ($v) { return !isset($v['scope']); })
+                                    ->thenInvalid('either "scope" or "base_uri" should be defined.')
+                                ->end()
+                                ->validate()
+                                    ->ifTrue(function ($v) { return isset($v['query']) && !isset($v['base_uri']); })
+                                    ->thenInvalid('"query" applies to "base_uri" but no base URI is defined.')
+                                ->end()
+                                ->children()
+                                    ->scalarNode('scope')
+                                        ->info('The regular expression that the request URL must match before adding the other options.')
+                                        ->cannotBeEmpty()
+                                    ->end()
+                                    ->scalarNode('base_uri')
+                                        ->info('The URI to resolve relative URLs, following rules in RFC 3985, section 2.')
```

Comment:
This is not the full purpose of this option. Well, it IS the full purpose if `scope` is not defined :p. But assuming it's not defined, this is both the base URI to use, but also for matching. Gotta find a way to "squeeze" that in. We could add an `info` on the parent `scopes` node?

Reply:
info updated